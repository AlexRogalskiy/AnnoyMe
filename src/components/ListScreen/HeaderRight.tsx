import React from "react";
import HeaderButtons, { HeaderButton } from "react-navigation-header-buttons";
import Icon from "react-native-vector-icons/MaterialIcons";

const HeaderButtonComponent = <T extends {}>(options: T) => (
  <HeaderButton {...options as any} IconComponent={Icon} iconSize={23} />
);

interface Props {
  onPress: () => void;
}

export default ({ onPress }: Props) => (
  <HeaderButtons HeaderButtonComponent={HeaderButtonComponent}>
    <HeaderButtons.Item
      title="Settings"
      iconName="settings"
      onPress={onPress}
    />
  </HeaderButtons>
);
