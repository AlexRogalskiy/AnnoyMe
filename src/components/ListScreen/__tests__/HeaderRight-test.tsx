import React from "react";
import renderer, { act } from "react-test-renderer";
import HeaderRight from "../HeaderRight";

it("HeadeRight renders button for navigator header", () => {
  const tree = renderer
    .create(<HeaderRight onPress={() => undefined} />)
    .toJSON();

  expect(tree).toMatchSnapshot("HeaderRight-button");
});

it("HeaderRight handles click", () => {
  const onPress = jest.fn(() => undefined);

  const component = renderer.create(<HeaderRight onPress={onPress} />).root;

  const button = component.findByProps({ testID: "header-button" });
  expect(button).toBeTruthy();

  act(() => button.props.onPress());
  expect(onPress.mock.calls).toEqual([[]]);
});
