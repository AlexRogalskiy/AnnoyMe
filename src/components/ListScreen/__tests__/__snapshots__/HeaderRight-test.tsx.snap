// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadeRight renders button for navigator header: HeaderRight-button 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "marginRight": 5,
      },
    ]
  }
>
  <View
    accessible={true}
    hitSlop={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID="header-button"
  >
    <View>
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 23,
            },
            Array [
              Object {
                "marginHorizontal": 11,
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î¢¸
      </Text>
    </View>
  </View>
</View>
`;
